.gitlab-ci.yml

```diff
@@ -0,0 +1,107 @@
+image: smk762/qt-ci:latest
+
+stages:
+  - build
+  - upload_zip
+  - upload_portable
+  - test
+  
+variables:
+  GIT_SUBMODULE_STRATEGY: recursive
+  DOCKER_DRIVER: overlay2
+  PIP_CACHE_DIR: "$CI_PROJECT_DIR/.cache/pip"
+  VCPKG_DEFAULT_BINARY_CACHE: "$CI_PROJECT_DIR/.cache/vcpkg"
+  QT_VERSION: "5.15.2"
+
+cache:
+  - key: $CI_COMMIT_REF_SLUG
+    paths:
+      - $CI_PROJECT_DIR/bundled/linux/atomicdex-desktop-linux-${CI_COMMIT_SHORT_SHA}.zip
+      - $CI_PROJECT_DIR/bundled/linux/atomicdex-desktop-${CI_COMMIT_SHORT_SHA}-x86_64.AppImage
+  - key: $CI_PROJECT_NAME
+    paths:
+      - "$CI_PROJECT_DIR/.cache/vcpkg"
+    
+
+linux:build:
+  stage: build
+  before_script:
+    - cd $CI_PROJECT_DIR
+    - mkdir -p .cache/pip
+    - mkdir -p .cache/vcpkg
+
+  script:
+    # Job Variables
+    - echo $CI_COMMIT_REF_SLUG
+    - echo $CI_COMMIT_AUTHOR
+    - echo $CI_COMMIT_BRANCH
+    - echo $CI_COMMIT_SHA
+    - echo $CI_COMMIT_SHORT_SHA
+    - echo $CI_PROJECT_PATH
+    - echo $CI_PROJECT_NAME
+    - echo $QT_VERSION
+
+    # Job Path/Env variables
+    - export DEBIAN_FRONTEND=noninteractive
+    - export QT_INSTALL_CMAKE_PATH=/opt/Qt/$QT_VERSION/gcc_64/lib/cmake
+    - export QT_ROOT=/opt/Qt/$QT_VERSION
+    - export Qt5_DIR=/opt/Qt/$QT_VERSION/gcc_64/lib/cmake/Qt5
+    - export PATH=/opt/Qt/$QT_VERSION/gcc_64/bin:$PATH
+    - export PATH=${PATH}:/cmake-3.19.0-rc3-Linux-x86_64/bin;
+    - export APPIMAGE_EXTRACT_AND_RUN=1
+
+    # Bootstrap Vcpkg
+    - cd $CI_PROJECT_DIR; ./ci_tools_atomic_dex/vcpkg-repo/bootstrap-vcpkg.sh
+
+    # STANDARD
+    #- cd $CI_PROJECT_DIR
+    #- rm -rf build || echo "no build folder to rm"
+    #- mkdir build
+    #- cd build
+    #- cmake -DCMAKE_BUILD_TYPE=Release ../
+    #- make -j atomicdex-desktop || echo "make atomicdex-desktop failed"
+    #- make install || echo "make install failed"
+    #- cd $CI_PROJECT_DIR
+    #- zip -r atomicdex-desktop-linux-built-${CI_COMMIT_SHORT_SHA}.zip build/bin   || echo "zip failed"
+    #- mv $CI_PROJECT_DIR/build/*.AppImage .
+    #- zip -r atomicdex-desktop-linux-built-${CI_COMMIT_SHORT_SHA}.AppImage.zip *.AppImage  || echo "zip failed"
+
+    # NINJA
+    - cd $CI_PROJECT_DIR
+    - rm -rf build_ninja || echo "no build_ninja folder to rm"
+    - mkdir build_ninja
+    - cd build_ninja
```

Comment:
this could be `mkdir build_ninja && cd $_`

Accoding to bash:

```
_         At  shell  startup,  set to the absolute pathname used to invoke
          the shell or shell script being executed as passed in the  envi‐
          ronment  or  argument  list.   Subsequently, expands to the last
          argument to the previous command, after expansion.  Also set  to
          the  full  pathname  used  to  invoke  each command executed and
          placed in the environment exported to that command.  When check‐
          ing  mail,  this  parameter holds the name of the mail file cur‐
          rently being checked."$_" is the last argument of the previous command.      
```

This follow the rules DRY: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself